### PR TITLE
Migrate to credentials

### PR DESCRIPTION
**Pull request by RobAWilkinson**
## Background
Looks like we lost part of this commit, opening PR to master to bring that in

Also looks like bad js tags were brought into the `application.html.erb` template.

Webpack handles all of that, no need to ever touch rails asset pipeline again 🎉

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -6,7 +6,7 @@ class User < ApplicationRecord
   # @see https://github.com/attr-encrypted/attr_encrypted#usage
   #
   attr_encrypted :personal_access_token,
-                 key: Rails.application.secrets.encryption_key,
+                 key: Rails.application.credentials.personal_access_token_key,
```
[the remark this answers is not in the document]
Comment (RobAWilkinson) [reply app/models/user.rb]: digging into it, the problem seems to lies with the fact that the key is a string in the credentials file, and needs to be a 32 byte array.  I've updated it so that it is saved as a 32 byte Hex string in the credentials file, then converted to a ByteArray in this line

File: app/models/user.rb [hunk 2]
```diff
@@ -4,9 +4,9 @@
 #
 class User < ApplicationRecord
   # @see https://github.com/attr-encrypted/attr_encrypted#usage
-  #
+  # @see https://github.com/attr-encrypted/attr_encrypted/issues/311
   attr_encrypted :personal_access_token,
-                 key: Rails.application.secrets.encryption_key,
+                 key: [Rails.application.credentials.personal_access_token_key].pack("H*"),
```
Comment (hpjaj) [inline app/models/user.rb]: Sorry, I was referring specifically to the _name_ of the key.  Instead of using `personal_access_token_key`, we'll want it to be `encryption_key`, b/c this is actually the encryption key that the `attr-encrypted` gem is using to encrypt the `personal_access_token` database column.

I just did a spike testing it on my side, and we are good to go with setting the `Rails.application.credentials.encryption_key` to a random string that is 256 bits.  I am using this for the test case:

```ruby
irb(main):009:0> key = SecureRandom.hex(16)
=> "a2635ff02e558385becc550145610818"
irb(main):010:0> key.length
=> 32
```

A string being 32 characters in length is in line with the 256 bits expectation (based on my quick research).  I then ran this to set the value in our credentials:

```
➜  magnifier git:(master) EDITOR="subl --wait" rails credentials:edit
```

And in the credentials file, I set:

```
encryption_key:  "a2635ff02e558385becc550145610818"  
```

Of course this is not our key :), just for test purposes.

Then, to test it with a user in the rails console, here is a user _without_ a PAT:

```
irb(main):004:0> ap harry
#<User:0x00007ff1edcdda48> {
  :id                                 => 1,
  :first_name                         => "Harry",
  :last_name                          => "Levine",
  :email                              => "harry@example.com",
  :encrypted_personal_access_token    => nil,
  :encrypted_personal_access_token_iv => nil,
  :created_at                         => Fri, 12 Oct 2018 02:39:37 UTC +00:00,
  :updated_at                         => Fri, 12 Oct 2018 02:39:37 UTC +00:00,
  :github_username                    => "hpjaj"
}
```

Then I add a PAT, confirm it gets encrypted, and is still accessible:

```
irb(main):005:0> harry.personal_access_token = 'some_secret_token'
=> "some_secret_token"
irb(main):006:0> harry.save
=> true
irb(main):007:0> ap harry
#<User:0x00007ff1edcdda48> {
  :id                                 => 1,
  :first_name                         => "Harry",
  :last_name                          => "Levine",
  :email                              => "harry@example.com",
  :encrypted_personal_access_token    => "c7y1r4Qg7gWeWgci0oJRNKS9MMTy3T+fAzinlmsTNxLz\n",
  :encrypted_personal_access_token_iv => "oBgaO9MXOGW4hwow\n",
  :created_at                         => Fri, 12 Oct 2018 02:39:37 UTC +00:00,
  :updated_at                         => Fri, 12 Oct 2018 18:46:36 UTC +00:00,
  :github_username                    => "hpjaj"
}
irb(main):008:0> harry.personal_access_token
=> "some_secret_token"
```

**TL;DR**, if you can just change line 9 to:

```ruby
Rails.application.credentials.encryption_key
```

I can take care of setting the values for us.
